util/shell/lex.go

```diff
@@ -374,27 +372,35 @@ func (sw *shellWord) processDollar() (string, error) {
 		return value, nil
 	}
 }
+
 func (sw *shellWord) processDollarShellOut() (string, error) {
 	var result bytes.Buffer
 
+	oldRawQuotes := sw.rawQuotes
+	sw.rawQuotes = true
+	defer func() { sw.rawQuotes = oldRawQuotes }()
+
 	seenParentheses := 1
 	escaped := false
 	for {
-		ch := sw.scanner.Next()
+		ch := sw.scanner.Peek()
 		if ch == scanner.EOF {
 			return "", errors.New("syntax error: missing ')'")
 		}
 		if escaped {
 			escaped = false
+			ch = sw.scanner.Next()
```

Comment:
not sure if it's more readable as `ch = sw.scanner.Next()` or `_ = sw.scanner.Next()`.

as `ch` is set above in `sw.scanner.Peek()`.

Reply:
Slight preference for `_ =`, but not a strong opinion.

Reply:
I should have looked earlier, but the upstream code already uses the `ch = sw.scanner.Next()` pattern, so i'll leave it as is.